runtime/doc/api.txt

```diff
@@ -626,6 +626,34 @@ nvim__stats()                                                  *nvim__stats()*
                 Return: ~
                     Map of various internal stats.
 
+                                                     *nvim_add_user_command()*
+nvim_add_user_command({name}, {command}, {*opts})
+                Create a new user command |user-commands|
+
+                {name} is the name of the new command. The name must begin
+                with an uppercase letter.
+
+                {command} is the replacement text to execute.
+
+                Example: >
+                   :call nvim_add_user_command('SayHello', 'echo "Hello world!"', {})
+                   :SayHello
+                   Hello world!
+<
+
+                Parameters: ~
+                    {name}     Name of the new user command. Must begin with
+                               an uppercase letter.
+                    {command}  Replacement command to execute when this user
+                               command is executed.
+                    {opts}     Optional parameters. See |command-attributes|
+                               for more details. To use boolean attributes
+                               (such as |:command-bang| or |:command-bar|) set
+                               the value to "true". When called from Lua,
+                               {opts} also accepts a "luaref" key that accepts
+                               a Lua function. When using a luaref, {command}
```

Comment:
I would make non-empty `{command}` an error, unless it is used for something else, like is it still printed as the definition?

Reply:
If we make `{command}` able to take a Lua func then this is a non-issue. Although it might be a good idea to be able to specify an optional description as an option.